src/endian.h

```diff
@@ -34,17 +34,18 @@ inline uint8_t u1be_read(const uint8_t* inBuffer)
 
 inline uint16_t u2be_read(const uint8_t* inBuffer)
 {
-	return ((uint16_t) u1be_read(inBuffer) << 8) | (u1be_read(inBuffer + 1));
+	// bitwise OR promotes types smaller than int to int
+	return (uint16_t) ((u1be_read(inBuffer) << 8) | (u1be_read(inBuffer + 1)));
```

Comment:
Hmm, maybe do
`STATIC_ASSERT(sizeof(uint32_tint) == sizeof(unsigned)` and explicitly cast both sides to unsigned?

Reply:
done, although my question is why not to cast directly to uint32_t?